packages/workflow/src/workflow.ts

```diff
@@ -165,7 +176,63 @@ async function scheduleActivityNextHandler({
         scheduleToCloseTimeout: msOptionalToTs(options.scheduleToCloseTimeout),
         startToCloseTimeout: msOptionalToTs(options.startToCloseTimeout),
         scheduleToStartTimeout: msOptionalToTs(options.scheduleToStartTimeout),
-        namespace: options.namespace,
+        headers,
+        cancellationType: options.cancellationType,
+      },
+    });
+  });
+}
+
+/**
+ * Push a scheduleActivity command into state accumulator and register completion
+ */
+async function scheduleLocalActivityNextHandler({
+  options,
+  args,
+  headers,
+  seq,
+  activityType,
+  attempt,
+  originalScheduleTime,
+}: LocalActivityInput): Promise<unknown> {
+  validateLocalActivityOptions(options);
+
+  return new Promise((resolve, reject) => {
+    const scope = CancellationScope.current();
+    if (scope.consideredCancelled) {
+      scope.cancelRequested.catch(reject);
+      return;
+    }
+    if (scope.cancellable) {
+      scope.cancelRequested.catch(() => {
+        if (!state.completions.activity.has(seq)) {
+          return; // Already resolved
+        }
+        state.pushCommand({
+          requestCancelLocalActivity: {
+            seq,
+          },
+        });
+      });
+    }
+    state.completions.activity.set(seq, {
+      resolve,
+      reject,
+    });
```

Comment:
This appears to be an exact dupe of the normal activity cancel handler stuff. Could be deduped.

Reply:
True, originally I had `completions.localActivity` but realized that LAs are resolved the same way as non-LAs.
I'll dedupe.

Reply:
The shared code is a bit harder to follow I'll push it anyways

Reply:
Ended up duplicating the code, I immediately created a bug when trying to dedupe.
I think this is the right approach for now.